### PR TITLE
Eslint - remove errors and some warnings

### PR DESCRIPTION
**Pull request by MariaRukat**

### CODE REVIEW

File: src/api/schoolDetails.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import useSWR from 'swr';
 import fetchData from './fetchData';
 
-export const fetchSchoolDetails = (path: string) => {
+export const fetchSchoolDetails = (path: string): Promise<any> => {
   return fetchData(path).then((res) => res.results[0]);
 };
 
-export const useSchoolDetails = (schoolId: number) => {
+export const useSchoolDetails = (schoolId: number): any => {
```
Comment (micorix) [inline src/api/schoolDetails.ts:8]: Don't SWR provide some type definitions to use?

File: src/components/sections/SchoolPage/SchoolHeader.tsx [hunk 2]
```diff
@@ -42,21 +42,19 @@ const SchoolHeader: FC<SchoolHeaderProps> = ({
   isPublic,
   schoolName,
   district,
-  description,
+  // description,
```
Comment (micorix) [inline src/components/sections/SchoolPage/SchoolHeader.tsx:45]: I think we can remove that. I'd rather say to avoid commenting out the code, especially not so important pieces of it, because most likely these comments will stay in the codebase for good.

File: src/components/Checkbox.tsx [hunk 3]
```diff
@@ -60,10 +60,9 @@ const CheckboxInput = styled.input`
   }
 `;
 
-const Checkbox = forwardRef(
-  (props: HTMLProps<HTMLInputElement>, ref: Ref<HTMLInputElement>) => (
-    <CheckboxInput {...props} type="checkbox" ref={ref} />
-  ),
-);
+const Checkbox = forwardRef((props: HTMLProps<HTMLInputElement>, ref: Ref<HTMLInputElement>) => (
+  // eslint-disable-next-line
```
Comment (micorix) [inline src/components/Checkbox.tsx:64]: I think we can disable `jsx-props-no-spreading`rule globally  in eslintrc.js.

File: src/components/Input.tsx [hunk 4]
```diff
@@ -57,12 +56,10 @@ interface InputWithAddonProps extends HTMLProps<HTMLInputElement> {
...
 
-export const InputWithAddon = React.forwardRef<
-  HTMLDivElement,
-  InputWithAddonProps
->((props, ref) => (
+export const InputWithAddon = forwardRef<HTMLDivElement, InputWithAddonProps>((props, ref) => (
   <InputWithAddonWrapper ref={ref} addonPosition={props.addonPosition}>
-    <span className={'addon'}>{props.addon}</span>
+    <span className="addon">{props.addon}</span>
+    {/* eslint-disable-next-line */}
```
Comment (micorix) [inline src/components/Input.tsx:62]: Same here

File: src/components/MobileFilters.tsx [hunk 5]
```diff
@@ -1,25 +1,29 @@
...
+import { Choice, FilterData } from '../data/filters';
+import styled from '../styling/styled';
 
-const FiltersButton = styled.button`
+const StyledButton = styled.button`
   border: none;
   background: transparent;
   display: flex;
   align-items: center;
+  outline: none;
```
Comment (micorix) [inline src/components/MobileFilters.tsx:14]: Let's leave style changes for another PR

File: src/components/MobileFilters.tsx [hunk 6]
```diff
@@ -1,25 +1,29 @@
...
   svg {
     color: ${(props) => props.theme.colors.primary};
     margin-right: 10px;
   }
 `;
+
 const MobileFiltersModal = styled.div<{ active: boolean }>`
   position: fixed;
-  top: 0;
+  top: 100px;
```
Comment (micorix) [inline src/components/MobileFilters.tsx:34]: ^

File: src/environments/environment.ts [hunk 7]
```diff
@@ -1,13 +1,15 @@
-import { environment as devEnvironment } from './environment.dev';
-import { environment as prodEnvironment } from './environment.prod';
-
-export interface Environment {
+interface Environment {
```
Comment (micorix) [inline src/environments/environment.ts:1]: Since this is the only file i this dir now, maybe we can move it to the higher level and remove the dir.

File: src/pages/SchoolsGridPage.tsx [hunk 8]
```diff
@@ -72,6 +72,14 @@ const LoadingCard = styled(Card)`
...
 
+interface ResultsProps {
+  schools: any;
+  error: any;
+  page: number;
+  count: number;
+  onPageChange: (page: number) => void;
+}
+
 const Results: FC<any> = ({ schools, error, page, count, onPageChange }) => {
```
Comment (micorix) [inline src/pages/SchoolsGridPage.tsx:83]: Why don't we use above interface here?

File: src/components/Footer.tsx [hunk 9]
```diff
@@ -1,7 +1,9 @@
 import React, { FC } from 'react';
-import styled from '../styling/styled';
 import { Link } from '@reach/router';
+import styled from '../styling/styled';
 import Container from './Container';
+import epf from '../assets/images/epf.png';
```
Comment (micorix) [inline src/components/Footer.tsx:5]: Maybe we can establish some rule to keep track of variables that holds images. Maybe we can add something like epfImage? But it's just a thought.

File: src/components/Footer.tsx [hunk 10]
```diff
@@ -1,7 +1,9 @@
...
+import kdp_logo from '../assets/images/kdp-logo.png';
```
Comment (micorix) [inline src/components/Footer.tsx:6]: Camel case pls

File: src/components/Map.tsx [hunk 11]
```diff
@@ -41,28 +41,25 @@ const Map: FC<MapProps> = (props) => {
...
   return (
-    <MapWrapper ref={props.innerRef}>
+    <MapWrapper ref={innerRef}>
       <MapElement ref={mapEl} />
     </MapWrapper>
   );
 };
 
-export default forwardRef<any, { onConfig?: Function }>((props, ref) => (
+export default forwardRef<any, { onConfig?: any }>((props, ref) => (
```
Comment (micorix) [inline src/components/Map.tsx:62]: Maybe we can use Pick utility type here?

File: src/components/Navbar.tsx [hunk 12]
```diff
@@ -100,6 +100,10 @@ const MobileBar = styled.div`
 
     @media (max-width: 780px) {
       display: block;
+      outline: none;
```
Comment (micorix) [inline src/components/Navbar.tsx:103]: Is this essential for eslint/some a11y rules to not throw errors? Maybe we can open another PR for style changes?

File: src/pages/Calculator.tsx [hunk 13]
```diff
@@ -243,7 +258,9 @@ const Calculator: FC<RouteComponentProps> = () => {
           <h2>
             Suma: <span>{(points.total ?? 0).toFixed()}</span>
           </h2>
-          <button onClick={resetForm}>Resetuj wynik</button>
+          <button type="button" onClick={resetForm}>
```
Comment (micorix) [inline src/pages/Calculator.tsx:261]: I remember there was some issue here. But maybe we can use `reset` type?

File: src/styling/fonts.ts [hunk 14]
```diff
@@ -1,4 +1,6 @@
-export const setupFonts = `
+/* eslint-disable global-require */
```
Comment (micorix) [inline src/styling/fonts.ts:1]: It was my bad. Is it possible to use imports instead of requires here?

**Review by micorix — CHANGES_REQUESTED** [review]
I noticed your work is not up to date with the master branch of the repo. Also, maybe it's on me, but I keep getting eslint error - "Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser".

**Comment by micorix** [comment]
@MariaRukat I also noticed when running your code locally that border-width when filters are selected is still the same. (It works on master).

File: src/api/schoolDetails.ts (hunk 1 above)
Comment (MariaRukat) [reply src/api/schoolDetails.ts:8]: Yes, but you have to add arguments, which describe the type of data (and we have another issue to do this)

File: src/components/sections/SchoolPage/SchoolHeader.tsx (hunk 2 above)
Comment (MariaRukat) [reply src/components/sections/SchoolPage/SchoolHeader.tsx:45]: ok!

File: src/components/Checkbox.tsx (hunk 3 above)
Comment (MariaRukat) [reply src/components/Checkbox.tsx:64]: good idea!

File: src/components/MobileFilters.tsx (hunk 6 above)
Comment (MariaRukat) [reply src/components/MobileFilters.tsx:34]: The filters menu is not visible in mobile version - you want to fix this in another issue?

File: src/environments/environment.ts (hunk 7 above)
Comment (MariaRukat) [reply src/environments/environment.ts:1]: But it's configured in this way in the project and moving it out causes problems

File: src/pages/SchoolsGridPage.tsx (hunk 8 above)
Comment (MariaRukat) [reply src/pages/SchoolsGridPage.tsx:83]: my mistake - I started to write this interface, but it needs more time and types of 'schools', so I wanted to move this to the moment when we will have type of 'schools' data

File: src/components/Footer.tsx (hunk 9 above)
Comment (MariaRukat) [reply src/components/Footer.tsx:5]: I did it in other places and here I forgot about it. These variables will have '_img' part at the and of the name

File: src/components/Navbar.tsx (hunk 12 above)
Comment (MariaRukat) [reply src/components/Navbar.tsx:103]: it's not essential for es-lint, but - it's connected with the need of changing 'span' to 'button' (because this is a clickable element) - so I needed to add some changes in style

File: src/pages/Calculator.tsx (hunk 13 above)
Comment (MariaRukat) [reply src/pages/Calculator.tsx:261]: Es-lint does not allow to 'reset' type - and I've found some discussions about this. Shortly - many people think that 'reset' type shouldn't be used. We can discuss this, but now I will leave it with ``type="button"``

File: src/styling/fonts.ts (hunk 14 above)
Comment (MariaRukat) [reply src/styling/fonts.ts:1]: I was trying to do this but I didn't find the way. So I just disabled the rule.

**Comment by MariaRukat** [comment]
@micorix I will check this error - I didn't see it before. And border-width in filters is working fine by me - but I will check it after updating repo with most recent changes

File: src/components/Footer.tsx (hunk 9 above)
Comment (micorix) [reply src/components/Footer.tsx:5]: Yeah, that's good! Can we stick to camel case though? :slightly_smiling_face:

File: src/styling/fonts.ts (hunk 14 above)
Comment (micorix) [reply src/styling/fonts.ts:1]: Ok. But I think we should get back to this later.

File: src/components/Navbar.tsx (hunk 12 above)
Comment (a-dabrowski) [reply src/components/Navbar.tsx:103]: http://www.outlinenone.com/

Comment (micorix) [reply src/components/Navbar.tsx:103]: Yeah, right. Thanks @a-dabrowski. @MariaRukat if we don't have styles that indicates focus status somehow maybe @karoli-na can help :)

File: src/components/Footer.tsx (hunk 9 above)
Comment (MariaRukat) [reply src/components/Footer.tsx:5]: yes :)

File: src/components/Navbar.tsx (hunk 12 above)
Comment (MariaRukat) [reply src/components/Navbar.tsx:103]: You are right, I forgot about it. So now I will leave outline and we can discuss it later with @karoli-na if we want to make our own styles for outline.

**Comment by micorix** [comment]
Hi. I'm still getting "parserOptions.project" error + border is not changing for me (IMO sth with colors, not border-width - it resizes properly). 
Will check it today again (propably my machine thing). 

I know that typing API calls is a subject for another PR but I think we can already fix `@typescript-eslint/explicit-module-boundary-types` & `@typescript-eslint/no-unused-vars` errors. What do you think?

**Comment by micorix** [comment]
Ok, style issues don't appear when I'm using private mode so I guess it's on me. 

BTW `parserOptions.project` error shows up only for files in `build` dir. Can you add that dir to `.eslintignore` or `ignorePatterns` pls?

**Comment by micorix** [comment]
Hi, just two small things:
1. I see you haven't auto-fixed some warnings yet. (`0 errors and 2 warnings potentially fixable with the `--fix` option.`)
2. I think we can fix `@typescript-eslint/explicit-module-boundary-types` & `@typescript-eslint/no-unused-vars errors` already in this PR. LMK what you think.

**Comment by MariaRukat** [comment]
Hi, I fixed those warnings you wrote about. 

So now we have to fix warnings from the rule `@typescript-eslint/no-explicit-any` - this is what we wanted to do in another PR.

**Comment by MariaRukat** [comment]
I added small fix :)

**Review by micorix — APPROVED** [review]
Looks good to me
